src/nvector/manyvector/nvector_manyvector.c

```diff
@@ -245,52 +240,45 @@ N_Vector N_VMake_MPIManyVector(MPI_Comm comm, sunindextype num_subvectors,
 N_Vector N_VNew_MPIManyVector(sunindextype num_subvectors, N_Vector* vec_array,
                               SUNContext sunctx)
 {
-  sunindextype i;
+  SUNFunctionBegin(sunctx);
   sunbooleantype nocommfound;
-
   MPI_Comm comm, vcomm;
-  int retval, comparison;
+  int comparison;
   N_Vector v = NULL;
 
   /* Check that all subvectors have identical MPI communicators (if present) */
   nocommfound = SUNTRUE;
-  vcomm       = MPI_COMM_NULL;
-  for (i = 0; i < num_subvectors; i++)
+  comm        = MPI_COMM_NULL;
+  for (sunindextype i = 0; i < num_subvectors; i++)
   {
     /* access MPI communicator for subvector i (vcomm);
        if none is present then continue to next subvector */
     vcomm = N_VGetCommunicator(vec_array[i]);
-
     if (vcomm == MPI_COMM_NULL) { continue; }
 
     /* if this is the first communicator, create a copy */
     if (nocommfound)
     {
       /* set comm to duplicate this first subvector communicator */
-      retval = MPI_Comm_dup(vcomm, &comm);
-      if (retval != MPI_SUCCESS) { return (NULL); }
-
+      SUNCheckMPICallNull(MPI_Comm_dup(vcomm, &comm));
       nocommfound = SUNFALSE;
 
       /* otherwise, verify that vcomm matches stored comm */
     }
     else
     {
-      retval = MPI_Comm_compare(vcomm, comm, &comparison);
-      if (retval != MPI_SUCCESS) { return (NULL); }
-
-      if ((comparison != MPI_IDENT) && (comparison != MPI_CONGRUENT))
-      {
-        return (NULL);
-      }
+      SUNCheckMPICallNoRet(MPI_Comm_compare(vcomm, comm, &comparison));
+      SUNCheckNull((comparison == MPI_IDENT) || (comparison == MPI_CONGRUENT),
+                   SUN_ERR_ARG_INCOMPATIBLE);
```

Comment:
It seems like this check should always happen otherwise there is no point in doing the `MPI_Comm_compare`. I will revert the `SUNCheckNull` to an if statement.  An alternative would be to add a `SUNCheckAlways`, but I that would have to wait for Phase III.